### PR TITLE
Add foreman statistics plugin

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Add foreman_statistics plugin

### CODE REVIEW

File: plugins/ruby-foreman-statistics/debian/rules [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+build:
+	cp cache/* /usr/share/foreman/vendor/cache/
+	cp $(PLUGIN).rb /usr/share/foreman/bundler.d/
+	cd /usr/share/foreman && ( \
+		/usr/bin/foreman-ruby /usr/bin/bundle install --local && \
+		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:apipie:cache[$(PLUGIN)] cache_part=resources \
+		OUT=/var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) RAILS_ENV=development DATABASE_URL=nulldb://nohost \
+		)
+	[ -e apipie-cache ] || mkdir apipie-cache/
+	cp -rp /var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) ./apipie-cache/
```
Comment (tbrisker) [inline plugins/ruby-foreman-statistics/debian/rules:23]: Looks like this step is failing for some reason:
```
20:38:43 cp -rp /var/lib/foreman/public/apipie-cache/plugin/foreman_statistics ./apipie-cache/
20:38:43 cp: cannot stat '/var/lib/foreman/public/apipie-cache/plugin/foreman_statistics': No such file or directory
20:38:43 debian/rules:15: recipe for target 'build' failed
20:38:43 make: *** [build] Error 1
20:38:43 dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2

File: plugins/ruby-foreman-statistics/debian/postinst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  else
+    $BUNDLE install --local --no-prune 2>&1 >> $LOGFILE
+  fi
+fi
+
+# DB migrate/seed after install
+if [ -f /usr/share/foreman/config/database.yml ]; then
+  if [ ! -z "${DEBUG}" ]; then
+    /usr/sbin/foreman-rake db:migrate || true
+    /usr/sbin/foreman-rake db:seed || true
```
Comment (tbrisker) [inline plugins/ruby-foreman-statistics/debian/postinst:45]: @ehelms didn't we drop these from the post installation? or did we only do that for rpms?

**Comment by ezr-ondrej** [comment]
:green_apple:

**Comment by mmoll** [comment]
Hrm, strange... shouldn't there be some assets to precompile (webpack or rails assets)?

**Comment by ezr-ondrej** [comment]
> Hrm, strange... shouldn't there be some assets to precompile (webpack or rails assets)?

There are webpack assets (no asset pipeline ones). Should I add the compilation somewhere? :thinking:

**Comment by mmoll** [comment]
I have no idea what's going on there, it seems that the precompilation tasks fails, perhaps because of `23:32:40 2020-06-18T23:32:40 [D|app|] There are pending migrations. Please run foreman-rake db:migrate.`

**Comment by tbrisker** [comment]
I don't think the pending migrations are blocking this, it's just informational, i got the task to run locally even with the pending migrations.
This line looks strange to me though:
`
02:32:40 2020-06-18T23:32:40 [D|app|] Loading foreman_statistics precompiled asset manifest from /usr/share/foreman/vendor/ruby/2.5.0/gems/foreman_statistics-0.1.1/public/assets/foreman_statistics/foreman_statistics.json
`
Could it be related to the sprockets upgrade? It looks like it's stopping the plugin asset compilation before it gets to the webpack compile. I am not seeing the same line when trying to run the same task locally.

File: plugins/ruby-foreman-statistics/debian/rules (hunk 1 above)
Comment (ezr-ondrej) [reply plugins/ruby-foreman-statistics/debian/rules:27]: this is fixed

**Comment by tbrisker** [comment]
Finally figured out what was happening - webpack folder was missing from the gem, https://github.com/theforeman/foreman_statistics/pull/15 should fix that.
This likely means that webpack didn't build properly for rpms as well.

**Comment by ezr-ondrej** [comment]
:crossed_fingers:

**Comment by ezr-ondrej** [comment]
:green_heart: :tada: :tada:

**Comment by tbrisker** [comment]
https://ci.theforeman.org/job/packaging_build_deb_plugin/510/

**Comment by ezr-ondrej** [comment]
Bump for RPM : #5578

**Pull request merged**